core/frontend/src/components/bridges/BridgeCreationDialog.vue

```diff
@@ -0,0 +1,180 @@
+            :rules="[validate_required_field, is_baudrate]"
+          />
+
+          <v-text-field
+            v-model="bridge.ip"
+            :rules="[validate_required_field, is_ip_address]"
+            label="IP address"
+          />
+
+          <v-text-field
+            v-model="bridge.udp_port"
+            :counter="50"
+            label="UDP port"
+            :rules="[validate_required_field, is_socket_port]"
+          />
+
+          <v-btn
+            color="success"
+            class="mr-4"
+            @click="createBridge"
+          >
+            Create
+          </v-btn>
+        </v-form>
+      </v-card-text>
+    </v-card>
+  </v-dialog>
+</template>
+
+<script lang="ts">
+import axios from 'axios'
+import Vue from 'vue'
+import { getModule } from 'vuex-module-decorators'
+
+import BridgetStore from '@/store/bridget'
+import NotificationStore from '@/store/notifications'
+import { Baudrate } from '@/types/common'
+import { bridget_service } from '@/types/frontend_services'
+import { LiveNotification, NotificationLevel } from '@/types/notifications'
+import { VForm } from '@/types/vuetify'
+import {
+  isBaudrate, isFilepath, isIntegerString, isIpAddress, isNotEmpty, isSocketPort,
+} from '@/utils/pattern_validators'
+
+const notification_store: NotificationStore = getModule(NotificationStore)
+const bridget_store: BridgetStore = getModule(BridgetStore)
+
+export default Vue.extend({
+  name: 'BridgeCreationDialog',
+  model: {
+    prop: 'show',
+    event: 'change',
+  },
+  props: {
+    show: {
+      type: Boolean,
+      default: false,
+    },
+  },
+
+  data() {
+    return {
+      bridge: {
+        serial_path: '',
+        baud: null as (number | null),
+        ip: '0.0.0.0',
+        udp_port: '15000',
+      },
+    }
+  },
+  computed: {
+    form(): VForm {
+      return this.$refs.form as VForm
+    },
+    available_baudrates(): {value: number, text: string}[] {
+      return Object.entries(Baudrate).map(
+        (baud) => ({ value: parseInt(baud[1], 10), text: baud[1] }),
+      )
+    },
+    available_serial_ports(): {value: string, text: string}[] {
+      return bridget_store.available_serial_ports.map(
+        (port) => ({ value: port, text: port }),
+      )
+    },
+    updating_serial_ports(): boolean {
+      return bridget_store.updating_serial_ports
+    },
+    serial_selector_label(): string {
+      return this.updating_serial_ports ? 'Fetching available serial ports...' : 'Serial port'
+    },
+  },
+  methods: {
+    validate_required_field(input: string | number): (true | string) {
+      const string_input = String(input)
+      return isNotEmpty(string_input) ? true : 'Required field.'
+    },
+    is_ip_address(input: string): (true | string) {
+      return isIpAddress(input) ? true : 'Invalid IP.'
+    },
+    is_path(input: string): (true | string) {
+      return isFilepath(input) ? true : 'Invalid path.'
+    },
+    is_socket_port(input: string): (true | string) {
```

Comment:
We should also check if it's inside the valid public port range, no ?

Reply:
Maybe. Not completely sure.